### DISCUSSION TITLE
Docker container property "quarkus.container-image.insecure" not used?

### DISCUSSION THREAD

**Discussion by zivkovic** [quarkusio/quarkus#37327] [Q&A] [answered]
Hi,

I currently have an insecure registry, which means I cannot push into it unless I add the specified registry into docker config like so (local machine config):
```
"insecure-registries": [
  "insecure_repo_url"
],
```

In Quarkus, I can see there is a configuration property to allow insecure registries found [here](https://quarkus.io/guides/container-image#quarkus-container-image_quarkus.container-image.insecure).

Since I build and push the image on github runner, the image fails to push, as the push command throws the x509 error (certificate invalid). As far as I understood, the above configuration property should allow me to push?

If I add the above "insecure-registries" property to my local machine and try to push, it allows me to push. 

I have also checked the code, to see where this "insecure" property is actually used, and I think [this](https://github.com/quarkusio/quarkus/blob/cd8965895d11c5b222abaea27902a4d67816bc33/extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java#L315) is the only use of the property. But again, this is used in JIB. Why is this not used for docker/other container providers as well? Is it not possible? If so, can we update the documentation to reflect that this property only works for JIB?

Thank you and best regards, Jan

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
@zivkovic if you know which docker command line arg to use, we can certainly add it

**Reply by zivkovic** [reply]
As far as I know, this cannot be done as a command line arg. It can only be done by configuring docker daemon using daemon.json and then restarting docker daemon. Restarting the docker daemon is not really an option for quarkus, therefor I think the only thing we can actually do is to update the documentation to let others (like me) know, that this setting only works for JIB. Maybe also include that each user can configure his own docker daemon using daemon.json as an alternative fix.

If you have any better ideas, I'm all ears.

Thank you.

**Reply by geoand** [reply]
Yeah, I think that makes sense.

Do you want to contribute that?

**Reply by zivkovic** [reply, marked as answer]
I'l try to do it this week when I get some time.
